### PR TITLE
[ATMO-991] fix email

### PR DESCRIPTION
**Pull request by cdosborn**
There's a corresponding tropo PR, which i need to link. But i'm grabbing :pizza: 

[Tropo PR](https://github.com/iPlantCollaborativeOpenSource/troposphere/pull/348)
[ATMO-991](https://pods.iplantcollaborative.org/jira/browse/ATMO-991)

### CODE REVIEW

File: core/email.py [hunk 1]
```diff
@@ -167,31 +167,16 @@ def request_info(request):
     return (user_agent, remote_ip, location, resolution)
 
 
-def send_email(subject, body, from_email, to, cc=None,
```
Comment (cdosborn) [inline core/email.py]: This chunk is just making a call to `send_email_task` with same arguments.

File: core/email.py [hunk 2]
```diff
@@ -250,14 +212,17 @@ def email_to_admin(
         username = 'Unknown'
     if request_tracker or not cc_user:
         # Send w/o the CC
-        return send_email(subject, body,
-                          from_email=email_address_str(username, user_email),
-                          to=[email_address_str(sendto, sendto_email)])
```
Comment (cdosborn) [inline core/email.py]: Pull out the differences in the two calls to `send_email`

File: core/templates/core/email/fingerprint.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Location: {{ location }}
```
Comment (cdosborn) [inline core/templates/core/email/fingerprint.html]: This is included in most emails that get sent out. I like using templates solely, much better than just tacking this onto an email body.

File: api/v2/views/email.py [hunk 4]
```diff
@@ -48,22 +47,75 @@ def _email(self, user, data):
...
+        message = render_to_string("instance_report.html", context=context)
+        email_success = email_admin(
+            self.request, subject, message, data=data)
+        email_response = {"email_sent": email_success}
+        if not email_success:
+            return Response(email_response, status=status.HTTP_400_BAD_REQUEST)
+        return Response(email_response, status=status.HTTP_200_OK)
 
 class FeedbackEmailViewSet(EmailViewSet):
-    required_keys = ["message", "user-interface"]
```
Comment (cdosborn) [inline api/v2/views/email.py]: Make atmo templates more lenient

**Comment by cdosborn** [comment]
@lenards this is a good sizeable chunk of work. I'd like this to be merged in, and I can create another PR for the other email feature. @Sang33ta asked nicely for the @7eaf4a9 fix a while ago.

**Comment by lenards** [comment]
@cdosborn - Okay.

File: api/v2/views/email.py [hunk 5]
```diff
@@ -13,11 +13,10 @@
 from api import permissions
 from api.v2.exceptions import failure_response
 
-from core.email import email_admin, resource_request_email
+from core.email import lookupEmail, resource_request_email, support_email, email_admin, request_data
```
Comment (lenards) [inline api/v2/views/email.py]: should `lookupEmail` be `lookup_email`?

File: core/email.py [hunk 6]
```diff
@@ -184,38 +167,16 @@ def request_info(request):
...
-        "fail_silently": fail_silently,
-        "html": html
-    }
-    send_email_task(*args, **kwargs)
-    return True
-
-
 def email_admin(request, subject, message, data=None,
-                cc_user=True, request_tracker=False):
+                cc_user=True, request_tracker=False, html=True):
```
Comment (lenards) [inline core/email.py]: 🎉

**Comment by lenards** [comment]
Connor - let's get Steve's approval on the Atmosphere PR first. 

@steve-gregory 📯

File: api/v2/views/email.py (hunk 5 above)
Comment (cdosborn) [inline api/v2/views/email.py]: @lenards lookupEmail is wrapping `iplantauth.protocol.ldap.lookupEmail`. It's correct but could have a better name..

File: api/v2/views/email.py [hunk 7]
```diff
@@ -48,22 +47,75 @@ def _email(self, user, data):
...
+            "provider": user.selected_identity.provider_uuid(),
+            "instance": instance,
+            "status": last_status
+        }
+
+        context.update(request_data(self.request))
+
+        message = render_to_string("instance_report.html", context=context)
+        email_success = email_admin(
+            self.request, subject, message, data=data)
```
Comment (steve-gregory) [inline api/v2/views/email.py]: add `request_tracker=True` here

File: api/v2/views/email.py [hunk 8]
```diff
@@ -117,11 +170,9 @@ class ResourceEmailViewSet(EmailViewSet):
...
     def _email(self, user, data):
-        quota = data.pop('quota')
-        reason = data.pop('reason')
         email_response = resource_request_email(
-            self.request, user.username, quota, reason)
-        if not email_response.get('email_sent', False):
-            return Response(email_response, status=status.HTTP_400_BAD_REQUEST)
-        else:
+            self.request, user.username, data["quota"], data["reason"])
+        if email_response.get('email_sent', False):
```
Comment (steve-gregory) [inline api/v2/views/email.py]: ensure the resource request email is sent to request_tracker

File: core/email.py [hunk 9]
```diff
@@ -549,14 +515,14 @@ def resource_request_email(request, username, new_resource, reason, options={}):
...
+        "quota": quota,
         "reason": reason,
         "url": request.build_absolute_uri(admin_url)
     }
-    body = render_to_string("core/email/resource_request.html",
-                            context=Context(context))
-    logger.info(body)
+    context.update(request_data(request))
+    body = render_to_string("resource_request.html", context=context)
+    success = email_admin(request, subject, body, cc_user=False)
```
Comment (steve-gregory) [inline core/email.py]: request_tracker=True

**Comment by steve-gregory** [comment]
LG2M! 👍

File: core/email.py [hunk 10]
```diff
@@ -521,10 +521,8 @@ def resource_request_email(request, username, quota, reason, options={}):
     }
     context.update(request_data(request))
     body = render_to_string("resource_request.html", context=context)
-    success = email_admin(request, subject, body, cc_user=False)
-
-    email_success = email_admin(request, subject, body, cc_user=False)
```
Comment (cdosborn) [inline core/email.py]: I was sending two emails earlier :fearful:

**Pull request merged**